[게시글 생성/조회/수정/삭제] 올레 미션 제출합니다.

jinwookh opened the pull request:
안녕하세요, 올레입니다.
세심한 리뷰 부탁드립니다!

dpudpu committed:
feat: 실습 구현

dpudpu committed:
docs: 요구사항 작성

dpudpu committed:
fix: static 상대경로 문제 절대경로로 변경해서  해결

jinwookh committed:
feat: devtools auto reload 추가

dpudpu committed:
feat: apply thymeleaf-dialect

dpudpu committed:
feat: apply thymeleaf-dialect

dpudpu committed:
feat: article 추가, 조회

jinwookh committed:
Merge remote-tracking branch 'dpu/dpudpu' into dpudpu

dpudpu committed:
feat: Article 목록 조회

dpudpu committed:
refactor: 필드인젝션-> 생성자인젝션으로 변경

dpudpu committed:
feat: article 수정

dpudpu committed:
feat: article 삭제

jinwookh committed:
Merge branch 'dpudpu' of https://github.com/dpudpu/jwp-blog into dpudpu

# Conflicts:
#	src/main/java/techcourse/myblog/web/ArticleController.java

jinwookh committed:
change welcome brown to welcome everyone

jinwookh committed:
add : creating article test case

jinwookh committed:
add : article-edit test case

jinwookh committed:
refactor: refactor code with convention

jinwookh committed:
refactor: delete comment code

jinwookh committed:
refactor: make method that check same id

article.hasSameId

jinwookh committed:
test: 게시글 생성 응답 코드 확인하는 테스트 작성

jinwookh committed:
test: 게시글 조회 요청 응답 코드 확인하는 테스트 작성

jinwookh committed:
test: 게시글 수정 페이지 응답 코드 확인하는 테스트 작성

jinwookh committed:
test: 게시글 수정 완료 후 페이지 응답 코드 확인하는 테스트 작성

jinwookh committed:
test: 존재하지 않는 게시글을 삭제하는 테스트 케이스 작성

jinwookh committed:
test: 기본 페이지 응답 코드 확인하는 테스트 작성

jinwookh committed:
test: articleRepository에 article이 저장되는지 확인하는 테스트 작성

jinwookh committed:
refactor: replace old app.css file to new app.css file

woowahanCU reviewed (changes requested):
안녕하세요 리뷰어 CU입니다. 리뷰가 많이 늦었네요. 미안해요 😭
몇가지 피드백 남겼으니 확인해보시고 궁금한 점 있으면 DM 남겨주세요.
주말 잘보내요~

jinwookh committed:
refactor: change data type of Article.id

long -> int

jinwookh committed:
refactor: change HelloWorldController location

jinwookh committed:
refactor: add code of finding index of Article, and set by that found index

before, equality of id and index is assumed to be true. However, they both be different in some case.

jinwookh committed:
refactor: remove setter of title, coverUrl, contents at Article class

instead use constructor to set those fields

jinwookh committed:
refactor: use createId method

jinwookh committed:
refactor: use stream api at findById method

jinwookh committed:
add: IllegalIdException that occurs when given id is not found

jinwookh committed:
refactor: make "/articles" to constant

constant name is ARTICLE_URL

jinwookh committed:
refactor: use RedirectView object

jinwookh committed:
refactor: test storage function of ArticleRepository class by comparing article objects

jinwookh committed:
pass HelloWorldController tests

jinwookh committed:
remove @ModelAttribute

jinwookh committed:
rename ArticleController methods name by convention

jinwookh committed:
use method reference at getMaxId method in ArticleRepository class

woowahanCU reviewed (approved):
피드백 반영 잘해주셨네요 👍 
질문주신 부분들 답변달아두었는데요. 확인하고 궁금한 부분 있으면 DM 남겨주세요.
수고하셨어요~ :)
